### PR TITLE
Reader Onboarding: List suggested sites in "step 2" modal

### PR DESCRIPTION
**Pull request by DustyReagan**
Related to https://github.com/Automattic/loop/issues/188

## Proposed Changes

* This PR pulls a list of suggested WPCOM blogs based on the users followed tags.
* It adds support for "curated" blogs defined in multiple files organized by category then tag.
* If a user follows a tag we have curated suggestions for, we display that above the API results.
* It introduces the concept of "weight." If a blog suggestion matches multiple tags, it should appear higher in the recommendation list.

Note: when you follow a site, it recommends more sites with an existing recommendation feature. We'll disable that feature in a new PR.

Screenshot 2024-10-10 at 5 32 50 PM

## Why are these changes being made?

* Push the Reader Onboarding project forward

## Testing Instructions

* Go to read?flags=reader/onboarding
* Follow some tags
* Click on "Discover and subscribe to sites you'll love"
* View the suggested sites

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/reader/onboarding/subscribe-modal/index.tsx [hunk 1]
```diff
@@ -1,17 +1,88 @@
...
+		queryKey: [ 'reader-onboarding-recommended-sites', followedTagSlugs ],
+		queryFn: () =>
+			wpcom.req.get(
+				{
+					path: `/read/tags/cards`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{
+					tags: followedTagSlugs,
+					site_recs_per_card: 6, // This does not seem to have any effect. 8 results are always returned.
```
Comment (DustyReagan) [inline client/reader/onboarding/subscribe-modal/index.tsx]: I copied this from @Addison-Stavlo [prototype](https://github.com/Automattic/wp-calypso/pull/84935/files#diff-cff253bd8ba10e74018ef69d5288ea8517a47186f54414732b16dd4e77cd4663R24), but it doesn't seem to change the number of results returned. Maybe I'm misunderstanding what `site_recs_per_card` does?

**Comment by davemart-in** [comment]
@DustyReagan this endpoint doesn't appear to be doing the trick. I selected just "Finance" and here are the sites it recommended:

CleanShot 2024-10-11 at 08 29 36@2x

I then selected just "Food" and it showed:

CleanShot 2024-10-11 at 08 30 48@2x

File: client/reader/onboarding/subscribe-modal/index.tsx (hunk 1 above)
Comment (Addison-Stavlo) [reply client/reader/onboarding/subscribe-modal/index.tsx]: > it doesn't seem to change the number of results returned. Maybe I'm misunderstanding what site_recs_per_card does?

I don't think its supposed to change the number of overall results (cards) returned, but how many recommendations are added to the `recommended_blogs` card type itself. Different recommendation cards are supposed to be filtered in between posts returned in the list, some for tags, blogs, and feeds.

The cards returned will have types (as you have seen below), standard `post` types, `interests_you_may_like` types (recommended tags), `recommended_blog` types, etc. And the param is supposed to change how many recommendations are on the `recommended_blog` type card (similar to `tag_recs_per_card` changes the number of tags listed on the card of type `interests_you_may_like`).

Im not currently seeing any of the `recommended_blog` type cards here, just the one tag card and a bunch of standard posts.
Screenshot 2024-10-11 at 10 04 38 AM

We may need to investigate the endpoint to figure out why the recommended_blogs card isn't being inserted?
`/wp-content/rest-api-plugins/endpoints/read-tags-cards.php` a comment near the top mentions that is should always return 8 posts with 2 recommendation cards added. 🤔

Comment (Addison-Stavlo) [reply client/reader/onboarding/subscribe-modal/index.tsx]: After investigating a little more, it looks like D159312-code (cc @eoigal) may be the culprit that has prevented the recommended blogs cards from appearing in the response.

It adds a line to the function getting the recommended blogs:
`$global_tag_recommendations_sites->enable_filter_new_or_empty_blogs();` that seems to be filtering everything out in my testing. If I comment this line out, I start to see recommended_blogs cards in the response again:

Screenshot 2024-10-11 at 10 19 04 AM

**Comment by eoigal** [comment]
The `recommended_blogs` should now be showing after I deployed a fix in D163715-code.

**Comment by holdercp** [comment]
I'm seeing the data coming back:
image

**Review by holdercp — COMMENTED** [review]
The code looks good to me.

I followed "Technology" and this was the result:
image

I then went and followed "Beer":
image

I did get a stuck placeholder that you can see there. Refreshing fixed the issue.

I did notice that the results were not deterministic and I was getting different suggestions every refresh. I'm guessing that's just how the endpoint works?

**Review by eoigal — APPROVED** [review]
This approach makes sense to me. Happy to iterate on it.

**Pull request merged**